doc/src/manual/performance-tips.md

```diff
@@ -355,109 +373,36 @@ foo (generic function with 2 methods)
 This keeps things simple, while allowing the compiler to generate optimized code in all cases.
 
 However, there are cases where you may need to declare different versions of the outer function
-for different element types of `a`. You could do it like this:
+for different element types or types of the `AbstractVector` of the field `a` in `MySimpleContainer`. You could do it like this:
 
-```
-function myfun(c::MySimpleContainer{Vector{T}}) where T<:AbstractFloat
-    ...
-end
-function myfun(c::MySimpleContainer{Vector{T}}) where T<:Integer
-    ...
-end
-```
-
-This works fine for `Vector{T}`, but we'd also have to write explicit versions for `UnitRange{T}`
```

Comment:
This whole thing seems redundant now since we can just do triangular dispatch. Changed to that instead.